Alternative bootstrap aggregation + new preferred order of ties

debe-kevin opened the pull request:
I have added the alternative aggregation scheme from the CLeaR paper. [see page 7, section 3.3 of the paper](https://proceedings.mlr.press/v236/debeire24a/debeire24a.pdf)
I have also considered the preferred order of ties for LPCMCI+, and I have a first version, but it will need further refinements. There are some limitations with the current implementation. For example, in case of ties between "-->" and "-?>" , the current implementation will assign a conflicting link "x-x", but in theory, "-?>" gives the least assertive claim between "-?>" and "-->" should be assigned. This is just one example, but it is the same for example for a tie between "o->" and "-->" or "<-o" and "<--". In general, when a tie arises between two link types (or more), the link type with the least assertive claim should be preferred.

debe-kevin committed:
Add alternative bootstrap aggregation. Add new preferred order of ties for LPCMCI+

jakobrunge commented on tigramite/pcmci_base.py:
```diff
@@ -946,46 +976,74 @@ def return_summary_results(results, conf_lev=0.9):
+            #In case of ties between other link types, a conflicting link "x-x" is assigned
+            preferred_order = [
+            "",       # No link (most conservative)
+            "o?o",    # No claim made (lag 0 only)
```
You can remove links from this list that involve "?" because LPCMCI's final output does not contain ?-middle marks.

debe-kevin reviewed (commented):
Right! Looks good :)

debe-kevin committed:
Update pcmci_base.py

removed links with ? in preferred_order list

jakobrunge committed:
Modify run_bootstrap_of to include aggregation parameter

Added aggregation parameter to run_bootstrap_of method and updated frequency calculations.
